piccolo_api/fastapi/endpoints.py

```diff
@@ -413,8 +413,13 @@ def modify_signature(
             ),
         ]
 
-        for field_name, _field in model.__fields__.items():
-            type_ = _field.outer_type_
+        for field_name, _field in model.model_fields.items():
+            # get type of field since field.outer_type_ is
+            # deprecated in Pydantic V2 (we use that for arrays
+            # in the OpenAPI docs). Using vars(), but we can
+            # also use _field.annotation.__dict__ (if that's better)
+            field_annotation = vars(_field.annotation)
+            type_ = field_annotation["__args__"][0]
```

Comment:
I'm not sure what the best approach is here. I know we used `outer_type` because if it's `list[int]` it returns `list`.

What does `_field.annotation` give us?

Reply:
For array field of strings `_field.annotation` is `typing.Optional[typing.List[str]]` but result is wrong

![without_args](https://github.com/piccolo-orm/piccolo_api/assets/30960668/9ea2896d-8b60-416c-80e4-7e00677900ff)


For array field of strings `field_annotation["__args__"][0]` is `typing.List[str]` which is correct

![with_args](https://github.com/piccolo-orm/piccolo_api/assets/30960668/f763ed06-4fd2-4c7b-9d9e-87ac20b3a906)

I don't know how to solve it otherwise, if you have any other idea feel free to write here.

Reply:
I just tried this and the `_get_type` function doesn't work as it should. The problem remains the same because the `Array` column does not display correct in the `OpenAPI` document filters. Now it's like this.

![without_args](https://github.com/piccolo-orm/piccolo_api/assets/30960668/9ea2896d-8b60-416c-80e4-7e00677900ff)

If I change the `_get_type` function to something like this
```python
def _get_type(type_: t.Type) -> t.Type:
    """
    Extract the inner type from an optional if necessary, otherwise return
    the type as is.
    """
    args = t.get_args(type_)
    return args[0]
```
the result is correct and looks like this.

![with_args](https://github.com/piccolo-orm/piccolo_api/assets/30960668/f763ed06-4fd2-4c7b-9d9e-87ac20b3a906)

Can you check that in case I'm wrong or doing something wrong.

Reply:
You're right - I can see the problem when trying to use the `Director.years_nominated` filter in the Swagger docs.

Will investigate.

Reply:
It's a bit odd, but `t.get_args` returns this:  `(typing.List[int], <class 'NoneType'>)` and I was expecting `(typing.List[int], None)`.

I need to work out what `NoneType` is vs `None`.

Reply:
So `type(None)` is `NoneType`. I've changed it to check for `None` and `NoneType`, and it should work.

Will add some tests.